src/amber/cli/templates/app/config/routes.cr

```diff
@@ -3,6 +3,7 @@ Amber::Server.configure do |app|
     # Plug is the method to use connect a pipe (middleware)
     # A plug accepts an instance of HTTP::Handler
     plug Amber::Pipe::PoweredByAmber.new
+    #plug Amber::Pipe::ClientIp.new(["X-Forwarded-For"])
```

Comment:
I like that this is disabled out of the box and also configured for the most common use case. Can we add a comment here or elsewhere that with some instructions about when this code should be activated?

Something like:

> Allows Amber to read the correct client information from headers set by http load balancers and proxies. Should remain disabled unless this application is going to be deployed behind a load balancer, otherwise it presents a security risk.

Reply:
No idea where a good position would be. I thought a description spanning more than one line wouldn't be appropriate to be placed above the pipe. You have any ideas where to put it, except in official docs?

Reply:
@robacarp I am somewhat leaning towards thinking that usage notes for each pipe (or any other piece of functionality) should be found in the documentation, rather than as comments in user visible/modifiable parts of configuration.

(It would certainly be more useful to have them in-place, agreed, but then various notes could be added to each pipe or feature, and it would be unreadable.)

Maybe once we document this in the documentation, just a link to the docs can be placed next to the pipe initialization.

Any opinions?

Reply:
Yeah, I agree. I saw your comment on the pipe after I wrote this, but I still think that it should have a note attached to it. Something terse which conveys that this thing is dangerous if you play with it outside of the right circumstances and points to the Guide (better?) or source code (at least) with more complete documentation on when and how it should be used.

@marksiemers Just because you said something in gitter about having written some of the guides, do you have any opinion on where this documentation should live?

Reply:
If it is about a pipe in a pipeline, it should be an addition to this document: https://amberframework.org/guides/getting-started/routing/pipelines.md#pipelines

There is a balance here - as that document may get too large and have to be split up, but for now I think we can just add content to it.

The actual change will happen here: https://github.com/amberframework/online-docs/blob/bec7377b2e980df73f5dfdd253aa5a227a679700/getting-started/routing/pipelines.md